src/views/home/components/token-selector/token-selector.view.tsx

```diff
@@ -0,0 +1,156 @@
+import { FC, MouseEvent, useEffect, useState } from "react";
+
+import Card from "src/views/shared/card/card.view";
+import Portal from "src/views/shared/portal/portal.view";
+import { Token, Chain } from "src/domain";
+import TokenAdder from "src/views/home/components/token-adder/token-adder.view";
+import useTokenSelectorStyles from "src/views/home/components/token-selector/token-selector.styles";
+import TokenList from "src/views/home/components/token-list/token-list.view";
+import TokenInfo from "src/views/home/components/token-info/token-info.view";
+
+type Screen =
+  | {
+      type: "token-list";
+    }
+  | {
+      type: "token-adder";
+      token: Token;
+    }
+  | {
+      type: "token-info";
+      token: Token;
+    };
+
+interface SelectedChains {
+  from: Chain;
+  to: Chain;
+}
+
+interface TokenSelectorProps {
+  account: string;
+  chains: SelectedChains;
+  tokens?: Token[];
+  onAddToken: (token: Token) => void;
+  onClose: () => void;
+  onRemoveToken: (token: Token) => void;
+  onSelectToken: (token: Token) => void;
+}
+
+const TokenSelector: FC<TokenSelectorProps> = ({
+  account,
+  chains,
+  tokens,
+  onAddToken,
+  onClose,
+  onRemoveToken,
+  onSelectToken,
+}) => {
+  const classes = useTokenSelectorStyles();
+
+  const [screen, setScreen] = useState<Screen>({
+    type: "token-list",
+  });
+
+  const handleOutsideClick = (event: MouseEvent) => {
+    if (event.target === event.currentTarget) {
+      onClose();
+    }
+  };
+
+  const handleNavigateToTokenAdder = (token: Token) => {
+    setScreen({
+      type: "token-adder",
+      token,
+    });
+  };
+
+  const handleNavigateToTokenDetails = (token: Token) => {
+    setScreen({
+      type: "token-info",
+      token,
+    });
+  };
+
+  const handleNavigateToTokenList = () => {
+    setScreen({
+      type: "token-list",
+    });
+  };
```

Comment:
I'd replace the "NavigateTo" part of these 3 handlers with "Show". The current names are confusing since we're not navigating anywhere, but showing a given screen in this same route.

Reply:
Well I don't agree with this. In fact, you are navigating through the `TokenSelector` screens, which are `TokenAdder`, `TokenInfo` and `TokenList`. Navigation is a more generic word than a route change.

Reply:
A matter of taste I guess.